### DISCUSSION TITLE
v4.3.3

### DISCUSSION THREAD

**Discussion by debugtalk** [httprunner/httprunner#1598] [0 - 公告（Announcements）]
**go version**

- feat: add `sleep_random` to sleep random seconds, with weight for multiple time ranges
- feat: input text with adb
- feat: add adb `screencap` sub command
- feat: add `IsAppInForeground` to check if the given package is in foreground
- feat: check if app is in foreground when step failed
- feat: add validator AssertAppInForeground and AssertAppNotInForeground
- feat: save screenshots of all steps including ocr and cv recognition process data
- fix: adb driver for TapFloat
- fix: stop logcat only when enabled
- fix: do not fail case when kill logcat error
- fix: take screenshot after each step
- fix: screencap compatibility for shell v1 and v2 protocol
- fix: display parsed url in html report
- fix: fast fail not closing the websocket connection
- fix #1467: failed to parse parameters with plugin functions
- fix #1549: avoid duplicate creating plugins

<hr /><em>This discussion was created from the release <a href='https://github.com/httprunner/httprunner/releases/tag/v4.3.3'>v4.3.3</a>.</em>

**Comment by cythf** [comment]
请问现在UI自动化这部分功能开放使用了吗？有具体的使用方法的文档吗？非常想尝试一下
